web-ui/src/main/resources/catalog/components/viewer/projectionSwitcher/ProjectionSwitcher.js

```diff
@@ -0,0 +1,175 @@
+              var oldProj = view.getProjection();
+              var newProj = ol.proj.get(projection);
+              
+              if(oldProj == newProj) {
+                return;
+              }
+              
+              var projectionConfig = {};
+              
+              $.each(gnViewerSettings.mapConfig.switcherProjectionList, function(i, config) {
+                if(config['code'] == projection) {
+                  projectionConfig = config;
+                }
+              });
+              
+              var newExtent = ol.proj.transformExtent(
+                  view.calculateExtent(scope.map.getSize()), oldProj, newProj);
+              
+              if(newProj.getExtent()) {
+                newExtent = newProj.getExtent();
+              }
+              
+              var newCenter = ol.proj.transform(scope.map.getView().getCenter(), oldProj, newProj);
+              
+              if(projectionConfig.center) {
+                newCenter = JSON.parse(projectionConfig.center);
+              }
+
+              var mapsConfig = {
+                projection: newProj,
+                extent: newExtent,
+                center: newCenter
+              };
+              
+              if (projectionConfig.resolutions) {
+                angular.extend(mapsConfig, {resolutions: JSON.parse(projectionConfig.resolutions)})
+              }
+
+              var newView = new ol.View(mapsConfig);
+
+              // Set the view
+              scope.map.setView(newView);
+
+              // Rearrange base layers to adapt (if possible) to new projection
+              var layersToRemove = [];
+
+              scope.map.getLayers().forEach(function(layer) {
+                if (layer.get("group") == 'Background layers' 
+                  || !layer.displayInLayerManager) {
+                  layersToRemove.push(layer);
+                }
+              });
+              
+              for (var i = 0; i < layersToRemove.length; i++) {
+                scope.map.removeLayer(layersToRemove[i]);
+              }
+
+
+              // Renew base layers
+              gnViewerSettings.bgLayers = [];
+              var layers = gnViewerSettings.mapConfig['map-viewer'].layers;
+              if (layers && layers.length) {
+                layers.forEach(function(layerInfo) {
+                  gnMap.createLayerFromProperties(layerInfo, scope.map)
+                    .then(function(layer) {
+                      if (layer) {
+                        layer.displayInLayerManager = false;
+                        layer.set("group", "Background layers");
+                        gnViewerSettings.bgLayers.push(layer);
+                      }
+                    });
+                });
+              }
+
+              // Reproject layers
+              scope.map.getLayers().forEach(function(layer) {
+                scope.changeLayerProjection(layer, oldProj, newProj);
+              });
```

Comment:
The whole logic here (delete/recreate/reproject layers) is kind of disturbing.
1. Layers are complex objects that may hold references to other objects; simply deleting them & recreating them from scratch is very risky in terms of side effects
2. Altering BG layers is also risky because the code that handles them is kind of messy (to say the least) and breaks very easily
2. I don't think it is needed anyway; as stated before, Openlayers will reproject layers on the fly when the view projection changes. When possible, layer sources should be recreated to use the new projection to have better quality (ie for WMS/WMTS layers), but that can be done without deleting layers

You may have seen this example, which is pretty close to the purpose of this PR:
http://openlayers.org/en/v3.16.0/examples/reprojection-by-code.html